internal/watcher/minio_test.go

```diff
@@ -40,7 +41,8 @@ func newWatcher(t *testing.T) (*miniredis.Miniredis, watcher.Watcher) {
 	}
 
 	var w watcher.Watcher
-	w, err = watcher.NewMinioWatcher(context.Background(), &config)
+	var logger logr.Logger
+	w, err = watcher.NewMinioWatcher(context.Background(), &config, logger)
```

Comment:
> Logging DLQ Size: I attempted to use logger.v2.info to log the number of items in the DLQ, but this approach consistently caused test failures. For now, I've removed this logging to prevent the panics, but this feels like a band-aid rather than a solution. We should explore why this is happening and how we can log the DLQ size without causing test failures.

The zero value of `loggger` panics when it's used. You should instead use `logger := logr.Discard()` or update logr to v1.2.4 where I believe they've changed the zero value so it doesn't panic.

Reply:
For now I have decided to use discard and will update logger in a separate pr